### PR TITLE
fix: truncate git status --porcelain output

### LINKED ISSUE #3672: [OwlBot] Postprocessor fails on a large monorepo when processing a large combined run
OwlBot post processing failed on this PR: https://github.com/googleapis/google-cloud-ruby/pull/18251/checks?check_run_id=6416435715 (see cloud build logs link on that page)

It appears from the logs that the postprocessing itself succeeded, but OwlBot was unable to push the resulting changes to the PR. Note: this is a large monorepo, and this change was a generator update that affected a large number of packages, and postprocessing involves moving files from a staging directory into the final package directory. Hence, the postprocessing commit would have been enormous, so I suspect this may be a scaling issue. The command that seems to have failed is in fact `git status --porcelain` which failed with `ENOBUFS`.

### PR DESCRIPTION
**Pull request by chingor13**
We don't actually need the entire list of files output from `git status --porcelain` -- we only check to see if there is anything that needs committing.

Fixes #3672

### CODE REVIEW

File: packages/owl-bot/src/git-utils.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Returns true if the specified directory has any git changes
+ * @param {string} cwd Directory to check
+ * @param {Cmd} cmd Optional command executor
+ * @returns {boolean} Returns true if there are any pending git changes
+ */
+export function hasGitChanges(cwd: string, cmd: Cmd = defaultCmd): boolean {
+  // `git status` --porcelain returns empty stdout when no changes are pending.
+  // We don't need the entire list, so only check to see if there's a single line
+  const status = cmd('git status . --porcelain | head -n 1', {cwd})
```
Comment (tmatsuo) [inline packages/owl-bot/src/git-utils.ts:28]: I think the status represents the return value of `head -n 1` which always succeeds.

**Review by tmatsuo — CHANGES_REQUESTED** [review]

File: packages/owl-bot/src/git-utils.ts (hunk 1 above)
Comment (chingor13) [reply packages/owl-bot/src/git-utils.ts:28]: This is the content of the `git status` command, not the exit status. I can rename the variable though

Comment (chingor13) [reply packages/owl-bot/src/git-utils.ts]: Renamed the variable to `output`

**Comment by chingor13** [comment]
I didn't see a way to have `git status --porcelain` return just an output code for if it has changes or not. `git diff` does have an `--exit-status` option, but doesn't suppress the diff (also misses added files).

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**